core/services/blockhashstore/delegate.go

```diff
@@ -152,14 +151,63 @@ func (d *Delegate) ServicesForSpec(jb job.Job) ([]job.ServiceCtx, error) {
 // AfterJobCreated satisfies the job.Delegate interface.
 func (d *Delegate) AfterJobCreated(spec job.Job) {}
 
+func (d *Delegate) OnCreateJob(jb job.Job, q pg.Queryer) error {
+	chain, err := d.getChainFromSpec(jb)
+	if err != nil {
+		d.logger.Error(err)
+		return nil
+	}
+
+	lp := chain.LogPoller()
```

Comment:
is this guaranteed to be non-nil? should .LogPoller() return an error under any cases (for example, not initialized?)

Reply:
Yeah, it will always be non-nil.  If log poller is disabled, then this will return a `LogPollerDisabled` implementation of `LogPoller`, whose methods all return `ErrDisabled`

Reply:
(Also, in this case it validates that log poller is enabled just before this, in `getChainFromSpec`, so shouldn't get here if it's disabled.)